context/src/main/java/io/grpc/Context.java

```diff
@@ -700,23 +700,22 @@ private CancellableContext(Context parent, Deadline deadline,
         ScheduledExecutorService scheduler) {
       super(parent, deriveDeadline(parent, deadline), true);
       if (DEADLINE_KEY.get(this) == deadline) {
-        final TimeoutException cause = new TimeoutException("context timed out");
         if (!deadline.isExpired()) {
           // The parent deadline was after the new deadline so we need to install a listener
           // on the new earlier deadline to trigger expiration for this context.
           pendingDeadline = deadline.runOnExpiration(new Runnable() {
             @Override
             public void run() {
               try {
-                cancel(cause);
+                cancel(new TimeoutException("context timed out"));
```

Comment:
If there were a lot of timeouts happening, this would slow things down.  Maybe this coudl be a helper function that laziliy creates the timeout?